utils/serviceClient.ts

```diff
@@ -45,76 +47,109 @@ export interface IServiceClient {
     context: Context
   ) => te.TaskEither<IResponseErrorInternal, LimitedProfile>;
   readonly submitMessageForUser: (
+    fiscalCode: FiscalCode,
     reqHeaders: NodeJS.Dict<string | ReadonlyArray<string>>,
     reqPayload: Response,
     context: Context
   ) => te.TaskEither<IResponseErrorInternal, Response>;
 }
 
+/**
+ * Given a pool of URLs, build a selector that retrieves one url after a provided fiscal code
+ * The purpose is to evenly distribute requests over the pool of different URLs
+ *
+ * @param pool the pool of urls
+ * @returns the selector function
+ */
+export const createPoolSelector = (
+  pool: ReadonlyArray<ValidUrl>
+): ((fiscalCode: FiscalCode) => ValidUrl) => {
+  const alphabet = Array.from({ length: 16 }).map((_, i) => i.toString(16));
+  // eslint-disable-next-line sonarjs/no-unused-collection
+  const chunks = Array<ReadonlyArray<string>>();
+  const chunkSize = Math.ceil(alphabet.length / pool.length);
+
+  // eslint-disable-next-line functional/no-let
+  for (let i = 0; i < pool.length; i++) {
+    // eslint-disable-next-line functional/immutable-data
+    chunks.push(alphabet.slice(i * chunkSize, i * chunkSize + chunkSize));
+  }
```

Comment:
We can use a more balanced chunking with
```suggestion
  const chunkSize = Math.floor(alphabet.length / pool.length);

  const chunks = pool.map((_, i, a) =>
    alphabet.slice(
      i * chunkSize,
      i === a.length - 1 ? alphabet.length : i * chunkSize + chunkSize
    )
  );
```
we have also an immutable array for chunks